File: taskManager/labfinal_210042111/Issue.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace labfinal_210042111
{
    public enum Priority { low, medium, high, critical }
    public enum Status { open, inProgress, underReview, resolved, closed }

    public class Issue
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public Priority priority { get; set; }
        public Status status { get; set; }
        public DateTime CreationDare { get; set; }
        public DateTime LastModificationDate { get; set; }
        public string AssignTo { get; set; }
        public List<string> Tags { get; set; }=new List<string>();
        public List<string> Comments { get; set; } = new List<string>();


        public Issue() { }


        public int GetId()
        {
            return Id;
        }
    }
}

File: TaskManager/Issue.java
import java.util.*;

enum Priority { LOW, MEDIUM, HIGH, CRITICAL }
enum Status { OPEN, IN_PROGRESS, UNDER_REVIEW, RESOLVED, CLOSED }

class Issue {
    private static int counter = 1;
    private int id;
    private String title;
    private String description;
    private Priority priority;
    private Status status;
    private Date creationDate;
    private Date lastModifiedDate;
    private String assignedTo;
    private List<String> tags = new ArrayList<>();
    private List<Comment> comments = new ArrayList<>();

    public Issue(String title, String description, Priority priority, String assignedTo) {
        this.id = counter++;
        this.title = title;
        this.description = description;
        this.priority = priority;
        this.status = Status.OPEN;
        this.creationDate = new Date();
        this.lastModifiedDate = new Date();
        this.assignedTo = assignedTo;
    }

    public void update(String title, String description, Priority priority, String assignedTo) {
        this.title = title;
        this.description = description;
        this.priority = priority;
        this.assignedTo = assignedTo;
        this.lastModifiedDate = new Date();
    }

    public void changeStatus(Status newStatus) {
        this.status = newStatus;
        this.lastModifiedDate = new Date();
    }

    public void addComment(Comment comment) {
        this.comments.add(comment);
    }

    public int getId() { return id; }
    public Status getStatus() { return status; }

    public String getDescription() {return description;}

}
